src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -692,3 +588,121 @@ def update_connectedk8s(cmd, instance, tags=None):
     with cmd.update_context(instance) as c:
         c.set_param('tags', tags)
     return instance
+
+# pylint:disable=unused-argument
+# pylint: disable=too-many-locals
+# pylint: disable=too-many-branches
+# pylint: disable=too-many-statements
+# pylint: disable=line-too-long
+
+
+def update_agents(cmd, client, resource_group_name, cluster_name, location=None, https_proxy="", http_proxy="", no_proxy="",
+                  kube_config=None, kube_context=None, no_wait=False, tags=None):
+    logger.warning("Ensure that you have the latest helm version installed before proceeding.")
+    logger.warning("This operation might take a while...\n")
+
+    # Setting user profile
+    profile = Profile(cli_ctx=cmd.cli_ctx)
+
+    # Setting kubeconfig
+    kube_config = set_kube_config(kube_config)
+
+    # Removing quotes from kubeconfig path. This is necessary for windows OS.
+    trim_kube_config(kube_config)
+
+    # Escaping comma, forward slash present in https proxy urls, needed for helm params.
+    https_proxy = escape_proxy_settings(https_proxy)
+
+    # Escaping comma, forward slash present in http proxy urls, needed for helm params.
+    http_proxy = escape_proxy_settings(http_proxy)
+
+    # Escaping comma, forward slash present in no proxy urls, needed for helm params.
+    no_proxy = escape_proxy_settings(no_proxy)
+
+    # Loading the kubeconfig file in kubernetes client configuration
+    try:
+        config.load_kube_config(config_file=kube_config, context=kube_context)
+    except Exception as e:
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=e, fault_type=consts.Load_Kubeconfig_Fault_Type,
+                                summary='Problem loading the kubeconfig file')
+        raise CLIError("Problem loading the kubeconfig file." + str(e))
+    configuration = kube_client.Configuration()
+
+    # Checking the connection to kubernetes cluster.
+    # This check was added to avoid large timeouts when connecting to AAD Enabled AKS clusters
+    # if the user had not logged in.
+    check_kube_connection(configuration)
+
+    # Get kubernetes cluster info for telemetry
+    kubernetes_version = get_server_version(configuration)
+    kubernetes_distro = get_kubernetes_distro(configuration)
+
+    kubernetes_properties = {
+        'Context.Default.AzureCLI.KubernetesVersion': kubernetes_version,
+        'Context.Default.AzureCLI.KubernetesDistro': kubernetes_distro
+    }
+    telemetry.add_extension_event('connectedk8s', kubernetes_properties)
```

Comment:
Do we need this for update as well? For create, it makes sense as we wanted to capture k8s version and distro. For update, we may not need these pivots on telemetry

Reply:
what about if the version has changed during update?